### PR TITLE
implement high level command `AUTH`

### PR DESCRIPTION
**Pull request by ArtemIsmagilov**
Hi. High-level authentication command is implemented. True, you have authentication at the moment of connection creation and that's great! I also want to be able to authenticate by a reserved user.
AUTH - https://valkey.io/commands/auth/
Test example - https://valkey.io/topics/acl/

### CODE REVIEW

File: redis/tests/test_basic.rs [hunk 1]
```diff
@@ -2015,4 +2015,23 @@ mod basic {
...
+        redis::cmd("ACL")
+            .arg("SETUSER")
+            .arg("alice")
+            .arg("on")
+            .arg(">p1pp0")
+            .arg("+ACL")
+            .exec(&mut con)
+            .unwrap();
+
+        assert_eq!(con.auth(Some("alice"), "p1pp0"), Ok(()));
```
Comment (nihohit) [inline redis/tests/test_basic.rs:2033]: please also test the None user scenario.

Comment (ArtemIsmagilov) [reply redis/tests/test_basic.rs:2076]: Done

Comment (nihohit) [reply redis/tests/test_basic.rs:2076]: Using the `None` option is cumbersome, and requires adding a meaningless type. This requires a more ergonomic solution.
